lite/valor_lite/classification/manager.py

```diff
@@ -0,0 +1,854 @@
+from collections import defaultdict
+from dataclasses import dataclass
+
+import numpy as np
+from numpy.typing import NDArray
+from tqdm import tqdm
+from valor_lite.classification.annotation import Classification
+from valor_lite.classification.computation import (
+    compute_confusion_matrix,
+    compute_metrics,
+)
+from valor_lite.classification.metric import (
+    F1,
+    ROCAUC,
+    Accuracy,
+    ConfusionMatrix,
+    Counts,
+    MetricType,
+    Precision,
+    Recall,
+    mROCAUC,
+)
+
+"""
+Usage
+-----
+
+manager = DataLoader()
+manager.add_data(
+    groundtruths=groundtruths,
+    predictions=predictions,
+)
+evaluator = manager.finalize()
+
+metrics = evaluator.evaluate()
+
+f1_metrics = metrics[MetricType.F1]
+accuracy_metrics = metrics[MetricType.Accuracy]
+
+filter_mask = evaluator.create_filter(datum_uids=["uid1", "uid2"])
+filtered_metrics = evaluator.evaluate(filter_mask=filter_mask)
+"""
+
+
+@dataclass
+class Filter:
+    indices: NDArray[np.int32]
+    label_metadata: NDArray[np.int32]
+    n_datums: int
+
+
+class Evaluator:
+    """
+    Classification Evaluator
+    """
+
+    def __init__(self):
+
+        # metadata
+        self.n_datums = 0
+        self.n_groundtruths = 0
+        self.n_predictions = 0
+        self.n_labels = 0
+
+        # datum reference
+        self.uid_to_index: dict[str, int] = dict()
+        self.index_to_uid: dict[int, str] = dict()
+
+        # label reference
+        self.label_to_index: dict[tuple[str, str], int] = dict()
+        self.index_to_label: dict[int, tuple[str, str]] = dict()
+
+        # label key reference
+        self.index_to_label_key: dict[int, str] = dict()
+        self.label_key_to_index: dict[str, int] = dict()
+        self.label_index_to_label_key_index: dict[int, int] = dict()
+
+        # computation caches
+        self._detailed_pairs = np.array([])
+        self._label_metadata = np.array([], dtype=np.int32)
+        self._label_metadata_per_datum = np.array([], dtype=np.int32)
+
+    @property
+    def ignored_prediction_labels(self) -> list[tuple[str, str]]:
+        """
+        Prediction labels that are not present in the ground truth set.
+        """
+        glabels = set(np.where(self._label_metadata[:, 0] > 0)[0])
+        plabels = set(np.where(self._label_metadata[:, 1] > 0)[0])
+        return [
+            self.index_to_label[label_id] for label_id in (plabels - glabels)
+        ]
+
+    @property
+    def missing_prediction_labels(self) -> list[tuple[str, str]]:
+        """
+        Ground truth labels that are not present in the prediction set.
+        """
+        glabels = set(np.where(self._label_metadata[:, 0] > 0)[0])
+        plabels = set(np.where(self._label_metadata[:, 1] > 0)[0])
+        return [
+            self.index_to_label[label_id] for label_id in (glabels - plabels)
+        ]
+
+    @property
+    def metadata(self) -> dict:
```

Comment:
add Returns to this docstring to make consistent with other docstrings in this file?

Reply:
Since its a property the one-liner describes the output.

It would be kinda redundant to say:

```
"""
Evaluation metadata.

Returns
---------
dict
    Evaluation metadat

```